File: ClientApp/src/app/models/usermodels.ts
import { NgModule } from '@angular/core';
import { FormBuilder, Validators, FormGroup, ReactiveFormsModule, FormsModule, FormArray } from '@angular/forms';
 //to update complex foreign key table put fake value in Id field of master table loke  0 in entity framework
export interface usermodels {
  username: string;
  userphone: string;
  address: userAddressmodels;
}
interface userAddressmodels {
  area: string;
  areaCode: number;
}
@NgModule({
  imports: [ReactiveFormsModule, FormsModule],
  exports: []
})
export class usermodelsform {
  modelForms: FormGroup = this.formBuilder.group({
    username: ["", Validators.required],
    userphone: ["", Validators.required],
    userAddress: this.formBuilder.array([this.modelAddressFormfunc()])
  });

  modelAddressFormfunc() {
    return this.formBuilder.group({
      userModelsId: [0], //to update complex foreign key table put fake value in Id field of master table loke  0 in entity framework
      area: '',
      areaCode: ''
    })
  }
  controlReturn() {
    const fa = this.modelForms.get('userAddress') as FormArray;
    return fa.controls;
  }
  pushInChild(val: any) {
   return  this.formBuilder.group({
      userModelsId: 0, //to update complex foreign key table put fake value in Id field of master table loke  0 in entity framework
      area: val['area'],
      areaCode: val['areaCode']
    })
  }
  getChildControl(mForm: FormGroup) {
    return <FormArray>mForm.controls['userAddress'];
  }
  constructor(private formBuilder: FormBuilder) { }

}

File: Models/userModels.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace angularcore2.Models
{
    public class userModels
    {
        public int Id { get; set; }
        public string username { get; set; }
        public string userphone { get; set; }

        public IList<userAddressModels> userAddress { get; set; }

    }
    public class userAddressModels
    {
        public int userAddressModelsId { get; set; }
        public int userModelsId { get; set; }
     
        public string area { get; set; }
        public int areaCode { get; set; }
    }
}
